packages/snjs/lib/services/protection_service.ts

```diff
@@ -170,6 +171,46 @@ export class SNProtectionService extends PureService<ProtectionEvent.SessionExpi
     }
   }
 
+  async getAuthorizedNotesForProtectedAction(
```

Comment:
Hmm yeah I'm struggling with the naming of this function. So this function authorized an action to be performed on an array of notes, and the output is the array of notes for which the action was authorized? And the output length can be different from the input length?

In that case maybe just `authorizeNotesForProtectedAction` would work?

Reply:
Or maybe actually `authorizeProtectedActionForNotes`

Reply:
Awesome! Yup that's the gist of it. I wasn't too fond of the name either, authorizeProtectedActionForNotes sounds good 🙂